draft-ietf-quic-transport.md

```diff
@@ -2497,16 +2497,25 @@ handshake is confirmed (see Section 4.1.2 of {{QUIC-TLS}}), an endpoint MUST
 send any CONNECTION_CLOSE frames in a 1-RTT packet.  However, prior to
 confirming the handshake, it is possible that more advanced packet protection
 keys are not available to the peer, so the frame MAY be replicated in a packet
-that uses a lower packet protection level.
+that uses a lower packet protection level.  More specifically:
 
-A client will always know whether the server has Handshake keys (see
-{{discard-initial}}), but it is possible that a server does not know whether the
-client has Handshake keys.  Under these circumstances, a server SHOULD send a
-CONNECTION_CLOSE frame in both Handshake and Initial packets to ensure that at
-least one of them is processable by the client.  Similarly, a peer might be
-unable to read 1-RTT packets, so an endpoint SHOULD send CONNECTION_CLOSE in
-Handshake and 1-RTT packets prior to confirming the handshake.  These packets
-can be coalesced into a single UDP datagram; see {{packet-coalesce}}.
+* A client will always know whether the server has Handshake keys (see
+  {{discard-initial}}), but it is possible that a server does not know whether
+  the client has Handshake keys.  Under these circumstances, a server SHOULD
+  send a CONNECTION_CLOSE frame in both Handshake and Initial packets to ensure
+  that at least one of them is processable by the client.
+
+* A client that sends CONNECTION_CLOSE in a 0-RTT packet cannot be assured of
+  the server has accepted 0-RTT and so sending a CONNECTION_CLOSE frame of type
+  0x1c in an Initial packet makes it more likely that the server can receive
+  the close signal, even if the application error code might not be received.
+
+* A peer might be unable to read 1-RTT packets, so an endpoint SHOULD send
```

Comment:
Suggestion to reorder this sentence: "Prior to confirming the handshake, a peer might be unable to process 1-RTT packets, so an endpoint SHOULD send CONNECTION_CLOSE in Handshake and 1-RTT packets."